test/functional/eval/input_spec.lua

```diff
@@ -196,6 +228,19 @@ describe('input()', function()
     eq('Vim(call):E118: Too many arguments for function: input',
        exc_exec('call input("prompt> ", "default", "file", "extra")'))
   end)
+  it('supports highlighting', function()
+    feed([[:call input({"highlight": "RainBowParens"})<CR>]])
+    wait()
+    feed('(())')
+    wait()
```

Comment:
May be a good idea to cache previous highlighting result and reuse it if prompt_id and text to highlight matches what was used previously.